### ISSUE TITLE
MSWord Paste into Body Text not working

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#456]
description to be added by @paregorios

**Comment by paregorios** [comment]
# Steps to reproduce

1. Create a new (or edit an existing) Page.
2. Switch focus to a Microsoft Word document, select and copy some text, and return to the browser window in which the "Add/Edit Page" form is displayed.
3. On the Add/Edit Page form, scroll down to the "Body Text" textbox and insert the text cursor in the field.
4. On the textbox toolbar, select the icon for "Paste from Word".
5. On the "Past from Word" dialog box, follow the instructions to "use ctrl-v (or cmd-v)on your keyboard to paste the text into the window."
6. Select the "insert" button on the "Paste from Word" dialog box.
7. Gaze in horror at the result:

![Screen Shot 2022-11-18 at 10 25 11 AM](https://user-images.githubusercontent.com/263285/202753138-d13a1add-4800-492f-8e01-5af58bdb8d43.png)

# Problem tested/confirmed in

- macOS Monterrey 12.6.1
   - Firefox 107.0 (64-bit)
   - @paregorios to add more

**Comment by paregorios** [comment]
What we want to get is the old behavior, i.e., just basic HTML character-level formatting comes through (bold, italic, underline, links).

**Comment by alecpm** [comment]
It probably makes sense to test out some of the [paste plugin options](https://www.tiny.cloud/docs-3x/reference/TinyMCE3x@Plugins/Plugin3x@paste/) for the TinyMCE version we are using to see if a more restrictive set of options will help resolve issues with the current version.

There are [some tips here](https://stackoverflow.com/questions/29997908/paste-from-word-to-tinymce) which seem to be oriented towards [TinyMCE versions 4 and 5,](https://www.tiny.cloud/docs-4x/plugins/paste/#paste_word_valid_elements) but they may be instructive if we need to do some customization. They could also be useful if/when we do upgrade the TinyMCE version (either using plone.app.widgets or upgrading to Plone 5.x).

Support for improved paste from Word on TinyMCE 4 and 5 is now provided by a [commercial add-on](https://www.tiny.cloud/docs-4x/enterprise/system-requirements/#wordcopypaste) which requires a [TinyMCE subscription](https://www.tiny.cloud/docs/tinymce/6/introduction-to-powerpaste/#getting-started-with-tinymce-pluginname), though at some point it seems to have been available as a standalone purchase for TinyMCE 4. I think that means it's unlikely that we'd get better support out of the box by upgrading Plone/TinyMCE.

**Comment by paregorios** [comment]
@alecpm I assume that testing these options out is a codebase/buildout tasks, yes? I don't see them surfaced in the control panel for the plugin.

**Comment by alecpm** [comment]
@paregorios Yeah, it's JS config of the widget, which probably requires a ZPT/DTML template customization (and there's no guarantee that there's a way to make the editor work with recent versions of Word). [This document](https://quintagroup.com/services/support/tutorials/plone-tinymce-paste-from-word) seems to imply that the only way to make this work reliably is to use Notepad as an intermediary for the paste (or better yet as the original rich text document source). Maybe that could be a "solution"?
